components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt

```diff
@@ -28,17 +30,27 @@ import kotlin.random.Random
 @Suppress("Deprecation")
 @Parcelize
 data class DownloadState(
-    val url: String,
-    val fileName: String? = null,
-    val contentType: String? = null,
-    val contentLength: Long? = null,
-    val userAgent: String? = null,
-    val destinationDirectory: String = Environment.DIRECTORY_DOWNLOADS,
-    val referrerUrl: String? = null,
-    val skipConfirmation: Boolean = false,
-    val id: Long = Random.nextLong(),
-    val sessionId: String? = null
+        val url: String,
+        val fileName: String? = null,
+        val contentType: String? = null,
+        val contentLength: Long? = null,
+        var currentBytesCopied: Long = 0,
+        @GuardedBy("context") var status: DownloadJobStatus = DownloadJobStatus.ACTIVE,
+        val userAgent: String? = null,
+        val destinationDirectory: String = Environment.DIRECTORY_DOWNLOADS,
+        val referrerUrl: String? = null,
+        val skipConfirmation: Boolean = false,
+        val id: Long = Random.nextLong(),
+        val sessionId: String? = null
 ) : Parcelable {
     val filePath: String get() =
         Environment.getExternalStoragePublicDirectory(destinationDirectory).path + "/" + fileName
 }
+
+enum class DownloadJobStatus {
+    ACTIVE,
+    PAUSED,
+    CANCELLED,
+    FAILED,
+    COMPLETED
+}
```

Comment:
We can move it inside of  `DownloadState` and rename it to `Status`. This way when we want to access will it be `DownloadState.Status`. Let's also add docs indicating its purpose and each status :)